cli/client.go

```diff
@@ -380,6 +387,77 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// CheckUpdateResponse holds the values used to hold release information.
+type CheckUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		infof(c.App.ErrWriter, "Failed to pare dateCompiled")
+		return err
+	}
+
+	if time.Since(dateCompiled) < time.Hour*24*7 {
```

Comment:
Worried about 1 week being too annoying -- Natalia suggested 6 in the ticket

Can we make `latest` versions warn after 6 weeks but `stable` versions continue the check if they are more than 1 week old?

Reply:
is there a way to tell which is installed?

Reply:
I think the presence or absence of the rconfig.Version can indicate that...

I would have to double-check all of the cases where the version is populated, but I think that is related.

Reply:
Actually, I can't think of a case where datecompiled and the version will not be simultaneously empty or populated. Nevermind.

Reply:
reviving this - I think 6 weeks for all installs is actually better (most customers will be on stable anyway so they're likely to get annoyed by it the most) and will simplify the logic

Reply:
So to confirm, check after the install is older than 6 weeks, then check weekly until the user performs a new install?

Reply:
make an api call every 3 days but the check/update message should happen on every invocation